update DNR samples to work in new DCC

patrickkettner opened the pull request:
fixes #1082

patrickkettner committed:
update DNR samples to work in new DCC

oliverdunk reviewed (approved):
LGTM, thanks for the fix!

hrmtsh2 commented:
Hi, I came here from the discussion on #1082 as I was having the same issue. However this fix also doesn't seem to change things, and somewhere down the line the changes were reverted?
I was trying to run this sample (in relation to my potential work on building an interactive tool to view changes in rules for DNR-bassed extensions for the GSoC program), but it doesn't seem to work.
As mentioned in #1082 by @oliverdunk, I tried to keep both `"main_frame"`, `"xmlhttprequest"` in the `resourceTypes` key to accommodate both sites with service workers and with main_frame, but to little avail. The only change I see is that the console message in `service_worker.js` now shows the message confirming the redirection (I assume the tabId of -1 indicates that it is not tab-specific?).
However, the redirection still doesn't occur. Kindly tell me where I am going wrong.

patrickkettner committed:
fix redirects for navigations served by the site's service worker

developer.chrome.com serves navigations through its service worker for
returning visitors, so there is no main_frame network request for these
rules to match. Match the service worker's xmlhttprequest as well, and
update the redirect destinations that have since moved.

patrickkettner commented:
@oliverdunk finally got back to this. I dug into why the reports here were inconsistent: developer.chrome.com serves navigations through its service worker for returning visitors, so there is no `main_frame` network request for DNR to match. Fresh profiles redirect fine, returning visitors get nothing, which lines up with what @hrmtsh2 saw (rule match logged with tabId -1, no actual redirect).

Matching `xmlhttprequest` as well (your suggestion from #1082) fixes it on current Chrome. I tested this branch on 149 with a fresh profile and again with the site's service worker registered, and both redirect now. Also updated the redirect destinations since the old ones have themselves become 301s, and added a README note about the service worker behavior.

Given the sample now needs that caveat explained to make sense, does it still seem worth merging, or would you rather we abandon it?

oliverdunk commented:
> Given the sample now needs that caveat explained to make sense, does it still seem worth merging, or would you rather we abandon it?

I'm still happy to land this given it is a simple change and gets the sample working more reliably. However, I think this is a confusing sample. Longer term my preference would be to just redirect some requests to example.com or similar. What do you think?

patrickkettner commented:
fine with me. ill assign a issue for me to redo the same soon.
